test/common/upstream/load_balancer_fuzz_base.cc

```diff
@@ -0,0 +1,208 @@
+
+  for (uint8_t locality = 0; locality < locality_indexes.size(); locality++) {
+    for (uint8_t index : localities[locality]) {
+      locality_indexes[locality].push_back(host_set.hosts_[index]);
+      locality_indexes_[index] = locality;
+    }
+  }
+
+  ENVOY_LOG_MISC(trace, "Added these hosts to locality 1: {}",
+                 absl::StrJoin(localities.at(0), " "));
+  ENVOY_LOG_MISC(trace, "Added these hosts to locality 2: {}",
+                 absl::StrJoin(localities.at(1), " "));
+  ENVOY_LOG_MISC(trace, "Added these hosts to locality 3: {}",
+                 absl::StrJoin(localities.at(2), " "));
+
+  host_set.hosts_per_locality_ = makeHostsPerLocality({locality_a, locality_b, locality_c});
+}
+
+// Initializes random and fixed host sets
+void LoadBalancerFuzzBase::initializeLbComponents(
+    const test::common::upstream::LoadBalancerTestCase& input) {
+  random_.initializeSeed(input.seed_for_prng());
+  for (uint8_t priority_of_host_set = 0;
+       priority_of_host_set < input.setup_priority_levels().size(); ++priority_of_host_set) {
+    initializeASingleHostSet(input.setup_priority_levels().at(priority_of_host_set),
+                             priority_of_host_set);
+  }
+  num_priority_levels_ = input.setup_priority_levels().size();
+}
+
+// Updating host sets is shared amongst all the load balancer tests. Since logically, we're just
+// setting the mock priority set to have certain values, and all load balancers interface with host
+// sets and their health statuses, this action maps to all load balancers.
+void LoadBalancerFuzzBase::updateHealthFlagsForAHostSet(const uint64_t host_priority,
+                                                        const uint32_t num_healthy_hosts,
+                                                        const uint32_t num_degraded_hosts,
+                                                        const uint32_t num_excluded_hosts,
+                                                        const std::string random_bytestring) {
+  const uint8_t priority_of_host_set = host_priority % num_priority_levels_;
+  ENVOY_LOG_MISC(trace, "Updating health flags for host set at priority: {}", priority_of_host_set);
+  MockHostSet& host_set = *priority_set_.getMockHostSet(priority_of_host_set);
+  // This downcast will not overflow because size is capped by port numbers
+  const uint32_t host_set_size = host_set.hosts_.size();
+  host_set.healthy_hosts_.clear();
+  host_set.degraded_hosts_.clear();
+  host_set.excluded_hosts_.clear();
+
+  Fuzz::ProperSubsetSelector subset_selector(random_bytestring);
+
+  const std::vector<std::vector<uint8_t>> subsets = subset_selector.constructSubsets(
+      {num_healthy_hosts, num_degraded_hosts, num_excluded_hosts}, host_set_size);
+
+  // Healthy hosts are first subset
+  for (uint8_t index : subsets.at(0)) {
+    host_set.healthy_hosts_.push_back(host_set.hosts_[index]);
+    // No health flags for healthy
```

Comment:
offline discussion about clearing the old flag value in order to make it healthy 
(and clear other vals after updating healthFlagSet or flag values)

Reply:
"other vals"? I'll look through codebase (Independence :)!!!)

Reply:
Actually you wanna just tell me what you mean by "other vals"

Reply:
Whoah bitwise or's (I've never seen this before) in clear/update: "void healthFlagClear(HealthFlag flag) override { health_flags_ &= ~enumToInt(flag); }
  bool healthFlagGet(HealthFlag flag) const override { return health_flags_ & enumToInt(flag); }
  void healthFlagSet(HealthFlag flag) final { health_flags_ |= enumToInt(flag); }"

Reply:
I'll have to iterate through original vectors to clear flags, hopefully this linear pass won't slow it down too much

Reply:
Another question is since the hosts updated don't all get put into a one of the healthy/degraded/excluded buckets, what do you do for the rest? I think that I'm gonna iterate through and set all their flags to healthy at the beginning, that seems logically correct to me (0 them out).

Reply:
Actually what I did was clear degraded and excluded flags at the very beginning because these are the ones that ever get set from previous iterations.